### PR TITLE
[ALLUXIO-3044] Use classload for all core UFS connectors

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-3044

### CODE REVIEW

File: assembly/server/pom.xml [hunk 1]
```diff
@@ -60,29 +60,9 @@
     </dependency>
     <dependency>
       <groupId>org.alluxio</groupId>
-      <artifactId>alluxio-underfs-gcs</artifactId>
-      <version>${project.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>org.alluxio</groupId>
       <artifactId>alluxio-underfs-local</artifactId>
```
Comment (aaudiber) [inline assembly/server/pom.xml]: should we do the same for local ufs?

Comment (apc999) [reply assembly/server/pom.xml]: that will break integration tests.
also local UFS brings no extra dependencies, thus I don't see a reason to class load local UFS

Comment (aaudiber) [reply assembly/server/pom.xml]: why does it break integration tests? can't they still depend on alluxio-underfs-local?

Comment (aaudiber) [reply assembly/server/pom.xml]: if it isn't too difficult, i'd like to classload all UFSes just to be consistent

**Comment by apc999** [comment]
@aaudiber PTAL

**Comment by aaudiber** [comment]
Sorry, above failure was my mistake messing with Jenkins

Jenkins, test this please

**Comment by aaudiber** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
@calvinjia I think the doc should still be fine except the wasb doc (no extra jars required on classpath). I will update wasb doc in a separate PR.

**Pull request merged**